src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,56 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Importance;
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.exceptions.UserNotFoundException;
+import com.tstepnik.planner.repository.TaskRepository;
+import com.tstepnik.planner.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.INCIDENTAL;
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        return taskRepository.findAll();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
```

Comment:
you are qurying DB eveyrtime user tries to add task. I dont like it. It is not optimal

Reply:
How can I do it better ?

Reply:
This way is optimal ?
` Authentication auth = SecurityContextHolder.getContext().getAuthentication();`

Reply:
Ok now I know is not, because auth doesn't return id.

Reply:
Dont store User in a Task, thats how. We would be good with just Id.

Reply:
This doesn't change anything because I still don't know how to extract LOGGED user Id.